src/globals.ts

```diff
@@ -134,10 +140,35 @@ export const ExecutionTags = (() => {
   // @ts-ignore
   const clear = () => (global.tags = []);
 
+  const getValue = (autoTagEvent: AutoTagEvent, innerKey) => {
+    let obj = autoTagEvent.event;
+    const eventByKey = autoTagEvent.keyToEventMap;
+    if (eventByKey && eventByKey[innerKey]) {
+      obj = eventByKey[innerKey];
+      return obj && { event: obj, keyToEventMap: eventByKey };
+    }
+    try {
+      if (obj && isString(obj) && obj[innerKey] === undefined) {
+        const parsedObj = JSON.parse(obj);
+        eventByKey[innerKey] = parsedObj[innerKey];
+        return parsedObj && { event: parsedObj[innerKey], keyToEventMap: eventByKey };
+      }
+    } catch (err) {}
+    if (obj && obj[innerKey]) {
+      eventByKey[innerKey] = obj[innerKey];
+      return { event: obj[innerKey], keyToEventMap: eventByKey };
+    }
+    return { event: undefined, keyToEventMap: eventByKey };
```

Comment:
Maybe its better to add a condition to handle when we got undefined

Reply:
then we return undefined